bpf/lib/lb.h

```diff
@@ -51,16 +56,75 @@ __BPF_MAP(cilium_lb4_services, BPF_MAP_TYPE_HASH, 0,
 	  sizeof(struct lb4_key), sizeof(struct lb4_service),
 	  PIN_GLOBAL_NS, CILIUM_LB_MAP_MAX_ENTRIES);
 
+__BPF_MAP(cilium_lb4_rr_seq, BPF_MAP_TYPE_HASH, 0,
+	  sizeof(struct lb4_key), sizeof(struct lb_sequence),
+	  PIN_GLOBAL_NS, CILIUM_LB_MAP_MAX_FE);
+
 #define REV_NAT_F_TUPLE_SADDR 1
+#ifdef LB_DEBUG
+#define cilium_trace_lb cilium_trace
+#else
+#define cilium_trace_lb(a, b, c, d)
+#endif
+
+static inline int lb_next_rr(struct __sk_buff *skb,
+			     struct lb_sequence *seq,
+			     __be16 hash)
+{
+	int slave = 0;
+	__u8 offset = hash % seq->count;
+
+	if (offset < LB_RR_MAX_SEQ) {
```

Comment:
Correct. For some reason it does not understand that if seq->count < LB_RR_MAX_SEQ, hash % seq->count will also be less than LB_RR_MAX_SEQ. So something like this would not work.

if (seq && seq->count > 0 && seq->count < LB_RR_MAX_SEQ)